r/R/dplyr-funcs-string.R

```diff
@@ -271,11 +285,13 @@ register_bindings_string_regex <- function() {
     if (opts$fixed) {
       out <- call_binding("endsWith", x = string, suffix = opts$pattern)
     } else {
-      out <- call_binding("grepl", pattern = paste0(opts$pattern, "$"), x = string, fixed = FALSE)
-    }
-
-    if (negate) {
-      out <- !out
+      out <- create_string_match_expr(
+        arrow_fun = "match_substring_regex",
+        string = string,
+        pattern = paste0(opts$pattern, "$"),
+        ignore_case = opts$ignore_case,
+        negate = negate
+      )
```

Comment:
Hmm, I don't think we have tests that would cover this (though we should!) but what would happen if someone called `str_ends("bar_foo", fixed("foo"), negate = TRUE)`? I think we might still need the separate `if (negate) {...}` block here

Reply:
Oh that's a great catch. I'll add a test and modify accordingly. It's probably worth taking that out of `create_string_match_expr()` and just doing it explicitly in the three functions that need it, as before.